scripts/lua/policies/security/oauth2.lua

```diff
@@ -55,37 +58,50 @@ function processWithRedis(red, securityObj)
   local token = {}
   local key = utils.concatStrings({"oauth:providers:", securityObj.provider, ":tokens:", accessToken})
   -- If we haven't cached the token go to the oauth provider
-  if not (redis.exists(red, key) == 1) then
-    token = exchange(accessToken, securityObj.provider)
-  else 
-    token = cjson.decode(redis.get(red, key))
+  if redis.exists(red, key) == 1 then
+    return cjson.decode(redis.get(red, key))
+  end
+ 
+  if securityObj.provider == 'facebook' or securityObj.provider == 'fake_facebook' then
```

Comment:
I'm going to have to hardcode a fake facebook anyways because we can't use the actual facebook provider since it makes a real http call, the only reason I have to put any code for facebook in here is because it is providing an additional header. This code is to do the actual looking for the tokens in the cache before it calls the provider.exchange() method. I could potentially take this out, but that means we're going to have to load provider code on each access rather than just going straight to the cache and loading this one module